docs/design/operator.md

```diff
@@ -0,0 +1,407 @@
+    leaderElection:
+      leaseDuration: 100
+      renewDeadline: 100
+      retryPeriod: 10
+    deployCSIAddons: true
+    cephFsKernelClient: true
+    cephFsKernelMountOptions: ms_mode=secure
+status:
+  phase: Succeeded
+  message: operator config successfully created
+```
+
+### CephCSIDriver CRD
+
+Manages the installation and configuration of CephFS, RBD, and NFS CSI drivers
+within namespaces. Allows customization of driver settings on a per-namespace
+basis. Only single instance of RBD/NFS/CephFS drivers can be created in a
+single namespace, If user wants to deploy new instance of RBD it need to be
+created in a new namespace. The operator need to ensure that no two CR'S are
+created in different namespace that are targetting same csi driver name.
+
+```yaml
+---
+kind: CephCSIDriver
+apiVersion: csi.ceph.io/v1alpha1
+metadata:
+  name: "<prefix>.<driver_type>.csi.ceph.com"
+  namespace: operator-namespace
+spec:
+  deploymentName: csi-cephfsplugin-provisioner
+  daemonsetName: csi-cephfsplugin
+  Spec:
+    fsGroupPolicy: File
+    encryption:
+      configMapRef:
+        name: encryption-config-map-name
+    plugin:
+      priorityClassName: system-node-critical
+      updateStrategy:
+        type: RollingUpdate
+        rollingUpdate:
+          maxUnavailable: 1
+      labels:
+        app: cephfs-plugin
+      annotations:
+        k8s.v1.cni.cncf.io/networks: macvlan-conf-1
+    provisioner:
+      labels:
+        app: ceph-fs-provisioner
+      annotations:
+        k8s.v1.cni.cncf.io/networks: macvlan-conf-1
+      provisionerReplicas: 2
+      leaderElection:
+        leaseDuration: 100
+        renewDeadline: 100
+        retryPeriod: 10
+      attachRequired: true
+    liveness:
+      metricsPort: 8000
+    deployCSIAddons: false
+  cephFsKernelClient: true
+  cephFsKernelMountOptions: ms_mode=secure
+status:
+  phase: Failed
+  message: Failed to create cephfs csi driver
+  reason: csi driver with same name already exists in the cluster
+```
+
+### CephCSICephConfig CRD
+
+Stores connection and configuration details for a single single Ceph cluster.
+Provides common connection and configuration information for multiple CSI
+drivers
+
+```yaml
+---
+kind: CephCSICephCluster
+apiVersion: csi.ceph.io/v1alpha1
+metadata:
+  name: ceph-cluster-1
+  namespace: operator-namespace
+spec:
+  monitors:
+  - 10.98.44.171:6789
+  - 10.98.44.172:6789
+  - 10.98.44.173:6789
+  ReadAffinity:
+    crushLocationLabels:
+    - kubernetes.io/hostname
+    - topology.kubernetes.io/region
+    - topology.kubernetes.io/zone
+  cephFS:
+    kernelMountOptions: readdir_max_bytes=1048576,norbytes
+    fuseMountOptions: debug
+  rbd:
+    mirrorDaemonCount: 2
+  cephConfig: |-
+    [global]
+    auth_cluster_required = none
+    auth_service_required = none
+    auth_client_required = none
+    rbd_validate_pool = false
+status: {}
+```
+
+### CephCSIConfig CRD
+
+Contains details about CephFS subvolume group or RADOS namespace and a
```

Comment:
Does this CRD replace the need to create a hash from the subvolume group name or rados namespace and add it to the storage class? Or how is it related?

Reply:
The name will be the hash what Rook generates today for subvolumegroup or radosnamespace and same will be added to the storageclass, Rook need to create this new CR when there is new subvolumegroup/radosnamespace  created